File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
class Solution
{
    public int solution(long n, long a, long b)
    {
        int count = 0;
        while(true){            
            
            if(a == b){
                return count;
            }
            
            count++;
                        
            a = (a % 2 == 1) ? (a+1)/2 : a/2;
            b = (b % 2 == 1) ? (b+1)/2 : b/2;            
        }
    }
}

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.py
import sys
import math
sys.setrecursionlimit(10 ** 9)

def recursive(a, b,  k):
    if a == b:
        return k
    
    return recursive(math.ceil(a / 2), math.ceil(b / 2), k + 1)

def solution(n,a,b):
    k = 1
    while a != b:
        a = math.ceil(a / 2)
        b = math.ceil(b / 2)
        k += 1
        
    return k - 1
        